cmd/strawberry/main.go

```diff
@@ -87,54 +87,63 @@ func (f FullValidatorInfo) ToMetadata() ([]byte, error) {
 // go run main.go -index 0
 func main() {
 	ctx := context.Background()
-	index := flag.String("index", "", "Validator Index")
+	index := flag.String("index", "", "Validator Configuration Index")
 	flag.Parse()
 
+	log := logger.NewConsoleLogger(logger.Options{Debug: true})
 	if *index == "" {
-		log.Fatal("listen address is required")
+		log.Fatal().Str("missing commandline flag", "-index").Msg("validator configuration index is required")
 	}
+
 	vs, err := loadFullValidatorInfos("test_validators.json")
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msgf("loading validator configuration failed")
 	}
+
 	i, err := strconv.Atoi(*index)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msgf("index parameter conversion failed")
 	}
+
 	if i < 0 || i >= len(vs) {
-		log.Fatalf("index %d out of bounds", i)
+		log.Fatal().Int("index", i).Msgf("validator configuration index out of bounds")
 	}
 	address := vs[i].IP
 	port := vs[i].Port
 	udpAddress, err := net.ResolveUDPAddr("udp", net.JoinHostPort(address, strconv.Itoa(port)))
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Str("address", address).Int("port", port).Err(err).Msgf("address resolve failed")
 	}
-	fmt.Printf("listening on: %v\n", address)
+
+	log.Info().Msgf("listening on: %v", address)
 	prv, err := hex.DecodeString(vs[i].Ed25519Prv)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msg("own private key decode failed")
 	}
 	pub, err := hex.DecodeString(vs[i].Ed25519Pub)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msg("own public key decode failed")
 	}
+
 	privateKey := ed25519.PrivateKey(prv)
 	publicKey := ed25519.PublicKey(pub)
-	vk := validator.ValidatorKeys{
+	vkeys := validator.ValidatorKeys{
```

Comment:
Variable name switched to avoid collision with `vk` in for loop below. Scope rules prevent problems, but I just didn't like the style.